### PR TITLE
[RBAC] failed grouprole permission migrations fix

### PR DESCRIPTION
**Pull request by jctanner**
https://issues.redhat.com/browse/AAP-31417

### CODE REVIEW

File: galaxy_ng/app/migrations/_dab_rbac.py [hunk 1]
```diff
@@ -39,18 +39,28 @@ def split_pulp_roles(apps, schema_editor):
...
                     new_data = {
                         'description': corerole.description,
                         'name': f'{corerole.name}_{pulp_assignment.content_type.model}'
                     }
                     new_role = Role(**new_data)
                     new_role.save()
+
+                    # add the permission back? ...
+                    for perm in pulp_assignment.role.permissions.all():
+                        new_role.permissions.add(perm)
```
Comment (AlanCoding) [inline galaxy_ng/app/migrations/_dab_rbac.py:59]: Yes, this was my oversight and this is the shape of the change I would have made. The only change I'd like is to skip any `perm` that has `perm.codename` not valid for the particular content type. That would be:

```python
from ansible_base.rbac.validators import permissions_allowed_for_role, combine_values

cls = pulp_assignment.content_type.model_class()
ct_codenames = combine_values(permissions_allowed_for_role(cls))
if perm.codename not in ct_codenames:
  continue
```

With test_app, example:

```
In [5]: combine_values(permissions_allowed_for_role(Namespace))
Out[5]: 
{'add_collectionimport',
 'change_collectionimport',
 'change_namespace',
 'delete_collectionimport',
 'delete_namespace',
 'view_collectionimport',
 'view_namespace'}
```

**Comment by jctanner** [comment]
With the latest commit, the namespace does retain it's groups ...

```
# curl -s -u admin:admin http://localhost:55001/api/galaxy/_ui/v1/namespaces/ | jq .
{
  "meta": {
    "count": 1
  },
  "links": {
    "first": "/api/galaxy/_ui/v1/namespaces/?limit=10&offset=0",
    "previous": null,
    "next": null,
    "last": "/api/galaxy/_ui/v1/namespaces/?limit=10&offset=0"
  },
  "data": [
    {
      "pulp_href": "/api/galaxy/pulp/api/v3/pulp_ansible/namespaces/1/",
      "id": 1,
      "name": "foospace",
      "company": "",
      "email": "",
      "avatar_url": "",
      "description": "",
      "groups": [
        {
          "id": 1,
          "name": "foogroup",
          "object_roles": [
            "galaxy.collection_namespace_owner_namespace"
          ]
        }
      ],
      "users": [],
      "related_fields": {},
      "metadata_sha256": "e03b8028831331e4b5226c154487b24d39955bbd71d4de89ffc5a85ebafe62de",
      "avatar_sha256": null
    }
  ]
}

```

However, nothing got copied over to the dab side ...

```
# curl -s -u admin:admin http://localhost:55001/api/galaxy/_ui/v2/role_team_assignments/ | jq .
{
  "count": 0,
  "next": null,
  "previous": null,
  "results": []
}
```

**Comment by AlanCoding** [comment]
The assignments should be copied later in `migrate_role_assignments`. There are a few logs from that which we might be able to look for in the output from what you ran. I would firstly look for:

> Data migration could not find team by name {group.name}

**Comment by AlanCoding** [comment]
Obvious question from last comment, can you see that a team exists for all groups after the migration?

The team model was created in the migration `0049_organization`. I'm struggling to find where we create teams from existing groups. There is a signal that does this, but signals don't run in migrations.

File: galaxy_ng/app/migrations/_dab_rbac.py [hunk 2]
```diff
@@ -39,18 +42,32 @@ def split_pulp_roles(apps, schema_editor):
     GroupRole = apps.get_model('core', 'GroupRole')
 
     for corerole in Role.objects.all():
+        print(f'ROLE {corerole} {corerole.name}')
         split_roles = {}
         for assignment_cls in (UserRole, GroupRole):
+            print(f'\t{assignment_cls}')
             for pulp_assignment in assignment_cls.objects.filter(role=corerole, content_type__isnull=False):
+                print(f'\t\t{assignment_cls} {pulp_assignment}')
```
Comment (AlanCoding) [inline galaxy_ng/app/migrations/_dab_rbac.py:50]: If you're going to leave these (which would be okay) then padding it with more detail would be helpful.

**Review by AlanCoding — APPROVED** [review]

File: galaxy_ng/app/migrations/_dab_rbac.py (hunk 2 above)
Comment (jctanner) [reply galaxy_ng/app/migrations/_dab_rbac.py:50]: not planning to leave any new prints ... just waiting for all tests to pass and all issues to get discovered before the final cleanup

**Comment by jctanner** [comment]
Failure from the certified sync workflow's migration step ...

```
FINISHED CREATING PERMISSIONS                                       
Traceback (most recent call last):                                                                                                       
  File "/usr/local/bin/pulpcore-manager", line 8, in <module>       
    sys.exit(manage())                                              
             ^^^^^^^^                                                                                                                    
  File "/usr/local/lib/python3.11/site-packages/pulpcore/app/manage.py", line 11, in manage                                              
    execute_from_command_line(sys.argv)                                                                                                                                                                                                                                           
  File "/usr/local/lib/python3.11/site-packages/django/core/management/__init__.py", line 442, in execute_from_command_line              
    utility.execute()                                                                                                                    
  File "/usr/local/lib/python3.11/site-packages/django/core/management/__init__.py", line 436, in execute                                
    self.fetch_command(subcommand).run_from_argv(self.argv)                                                                              
  File "/usr/local/lib/python3.11/site-packages/django/core/management/base.py", line 412, in run_from_argv                              
    self.execute(*args, **cmd_options)                                                                                                   
  File "/usr/local/lib/python3.11/site-packages/django/core/management/base.py", line 458, in execute                                    
    output = self.handle(*args, **options)           
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^                                                                                               
  File "/usr/local/lib/python3.11/site-packages/django/core/management/base.py", line 106, in wrapper                                    
    res = handle_func(*args, **kwargs)                      
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^                                                                                                   
  File "/usr/local/lib/python3.11/site-packages/django/core/management/commands/migrate.py", line 356, in handle                         
    post_migrate_state = executor.migrate(                
                         ^^^^^^^^^^^^^^^^^                                                                                               
  File "/usr/local/lib/python3.11/site-packages/django/db/migrations/executor.py", line 135, in migrate                                  
    state = self._migrate_all_forwards(            
            ^^^^^^^^^^^^^^^^^^^^^^^^^^^                                                                                                  
  File "/usr/local/lib/python3.11/site-packages/django/db/migrations/executor.py", line 167, in _migrate_all_forwards                    
    state = self.apply_migration(                                   
            ^^^^^^^^^^^^^^^^^^^^^                                                                                                        
  File "/usr/local/lib/python3.11/site-packages/django/db/migrations/executor.py", line 252, in apply_migration                          
    state = migration.apply(state, schema_editor)                  
            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^                                                                                        
  File "/usr/local/lib/python3.11/site-packages/django/db/migrations/migration.py", line 132, in apply                                   
    operation.database_forwards(                                                                                                         
  File "/usr/local/lib/python3.11/site-packages/django/db/migrations/operations/special.py", line 193, in database_forwards              
    self.code(from_state.apps, schema_editor)                                                                                            
  File "/src/galaxy_ng/galaxy_ng/app/migrations/_dab_rbac.py", line 57, in split_pulp_roles                                              
    ct_codenames = combine_values(permissions_allowed_for_role(cls))                                                                     
                                  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^                                                                      
  File "/usr/local/lib/python3.11/site-packages/ansible_base/rbac/validators.py", line 51, in permissions_allowed_for_role               
    raise ValidationError(f'Django-ansible-base RBAC does not track permissions for model {cls._meta.model_name}')                                                                                                                                                                
rest_framework.exceptions.ValidationError: [ErrorDetail(string='Django-ansible-base RBAC does not track permissions for model synclist', code='invalid')]
```

**Review by rochacbruno — APPROVED** [review]

**Pull request merged**